workspace/resource_directory.go

```diff
@@ -44,7 +48,12 @@ func ResourceDirectory() *schema.Resource {
 			d.SetId("")
 			return fmt.Errorf("different object type, %s, on this path other than a directory", objectStatus.ObjectType)
 		}
-		return common.StructToData(objectStatus, s, d)
+		err = common.StructToData(objectStatus, s, d)
+		if err != nil {
+			return err
+		}
+		d.Set("workspace_path", "/Workspace"+d.Id())
```

Comment:
Is there any scenario in which this is called before the Id is set for this resource? Maybe we just use `.Path` instead like we do for repos?

Reply:
No, for Read operation `d.Id()` is always set - see `objectStatus, err := notebooksAPI.Read(d.Id())` earlier... Also, for directories `path == d.Id()`